e2e/tests/uniswap_trade_test.rs

```diff
@@ -0,0 +1,186 @@
+use contracts::{ERC20Mintable, GPv2Settlement, UniswapV2Factory, UniswapV2Router02};
+use ethcontract::prelude::{Account, Address, Http, PrivateKey, Web3, U256};
+use model::{DomainSeparator, OrderCreationBuilder, OrderKind};
+use orderbook::orderbook::OrderBook;
+use secp256k1::SecretKey;
+use serde_json::json;
+use std::{str::FromStr, sync::Arc};
+use web3::signing::SecretKeyRef;
+
+const TRADER_A_PK: [u8; 32] = [
+    0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1,
+];
+const TRADER_B_PK: [u8; 32] = [
+    0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 2,
+];
+
+const NODE_HOST: &str = "http://localhost:8545";
+const API_HOST: &str = "http://localhost:8080";
+const ORDER_PLACEMENT_ENDPOINT: &str = "/api/v1/orders/";
+
+#[tokio::test]
+async fn test_with_ganache() {
```

Comment:
Because the test requires ganache I prefer if it didn't ran on `cargo test` in the crate root. Could achieve this by using  default-members in the workspace Cargo.toml or we could make the test a real binary target instead of a test. Or we could create an "integration test" as https://doc.rust-lang.org/stable/book/ch11-03-test-organization.html#integration-tests described. I have never tried. I also don't mind having whatever is the simplest solution first and doing more work in a later PR.

Reply:
Good point. I made everything but e2e a default-member. This achieves that it isn't run on `cargo test`. On travis we run `cargo test --workspace` so it's covered.